src/app/controllers/sap.php

```diff
@@ -68,6 +68,53 @@ public function logout() {
         $this->sap_auth->logout();
     }
 
+    public function verify($hash="") {
+        $this->load_library('http_lib');
+        if ($hash == "") {
+            $this->http_lib->response_code(404);
+        }
+        $this->load_model('sap_model');
+        $user = $this->sap_model->verify_hash($hash);
+        if (! $user) {
+            $this->http_lib->response_code(404);
```

Comment:
I skimmed through the https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_Error. There seem to be some that sound right, but they're about HTTP headers, or in general, don't match the kind of case we're dealing with here. :/ I might be wrong, but can't find one right now.